casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala

```diff
@@ -246,9 +246,9 @@ class Initializing[F[_]
         .map(_.latestBlockHash)
         .toSet
 
-      // Add sorted DAG in order from oldest to approved block
+      // Add sorted DAG in order from approved block to oldest
       minHeight <- getMinBlockHeight
-      _ <- heightMap.flatMap(_._2).toList.traverse_ { hash =>
+      _ <- heightMap.flatMap(_._2).toList.reverse.traverse_ { hash =>
```

Comment:
We have to add LFB first, because each `DagStore.insert` returns DagRepresentation, which requires having finalized block in DagStore. Also each insert checks consistency of the DAG (heights should be continuous)
https://github.com/rchain/rchain/blob/56c35c0ecbdc2b5ba0d5cd5a02c960fb7d7d4e74/block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockMetadataStore.scala#L186-L192
So fix is to add blocks in reverse order - down from LFB.